src/level_four/core/controller/send_transaction_controller_solana.c

```diff
@@ -175,7 +175,31 @@ void send_transaction_controller_solana() {
     } break;
 
     case SEND_TXN_TAP_CARD_SEND_CMD_SOLANA: {
-      flow_level.level_three = SEND_TXN_READ_DEVICE_SHARE_SOLANA;
+      flow_level.level_three = SEND_TXN_UPDATE_BLOCKHASH_SOLANA;
+      uint8_t arr[1]         = {1};
+      transmit_data_to_app(SEND_TXN_PRE_SIGNING_DATA, arr, sizeof(arr));
```

Comment:
Would prefer to move this command response to a task before the pin entry.
You can check the flow_status as a trigger for sending blockhash

Reply:
Also `SEND_TXN_PRE_SIGNING_DATA` is not the right response code. Device should send response to the previous command id `SEND_TXN_REQ_UNSIGNED_TXN`

Reply:
Updated the command Id but couldn't move the send command as the desktop still expects pin / passphrase/ card tap rejection